File: GestaoAcademicaWeb_old/app/Classes/Disciplina.php
<?php
namespace \app\Classes;
/**
 *
 */
 class Disciplina{

private $id_disciplina;
private $nome;
private $disciplsigla;
private $creditos;
private $descricao;
private $cargahoraria;
private $modulopertencente;
private $disciplrequisito;
private $id_plano_ensino;
private $id_professor;
public function __construct($id_disciplina = 0,$nome= "" ,$disciplsigla= "" ,
                            $creditos = 0,$descricao= "" ,$cargahoraria = 0,
                            $modulopertencente = 0,$disciplrequisito= "" ,
                            $id_plano_ensino = 0,$id_professor = 0){
$this->id_disciplina = $id_disciplina;
$this->nome = $nome;
$this->disciplsigla = $disciplsigla;
$this->creditos = $creditos;
$this->descricao = $descricao;
$this->cargahoraria = $cargahoraria;
$this->modulopertencente = $modulopertencente;
$this->disciplrequisito = $disciplrequisito;
$this->id_plano_ensino = $id_plano_ensino;
$this->id_professor = $id_professor;

}

public static function construct($array){
$obj = new Disciplina();
$obj->setId_disciplina( $array['id_disciplina']);
$obj->setNome( $array['nome']);
$obj->setDisciplsigla( $array['disciplsigla']);
$obj->setCreditos( $array['creditos']);
$obj->setDescricao( $array['descricao']);
$obj->setCargahoraria( $array['cargahoraria']);
$obj->setModulopertencente( $array['modulopertencente']);
$obj->setDisciplrequisito( $array['disciplrequisito']);
$obj->setId_plano_ensino( $array['id_plano_ensino']);
$obj->setId_professor( $array['id_professor']);
return $obj;

}

public function getId_disciplina(){
return $this->id_disciplina;
}

public function setId_disciplina($id_disciplina){
 $this->id_disciplina=$id_disciplina;
}

public function getNome(){
return $this->nome;
}

public function setNome($nome){
 $this->nome=$nome;
}

public function getDisciplsigla(){
return $this->disciplsigla;
}

public function setDisciplsigla($disciplsigla){
 $this->disciplsigla=$disciplsigla;
}

public function getCreditos(){
return $this->creditos;
}

public function setCreditos($creditos){
 $this->creditos=$creditos;
}

public function getDescricao(){
return $this->descricao;
}

public function setDescricao($descricao){
 $this->descricao=$descricao;
}

public function getCargahoraria(){
return $this->cargahoraria;
}

public function setCargahoraria($cargahoraria){
 $this->cargahoraria=$cargahoraria;
}

public function getModulopertencente(){
return $this->modulopertencente;
}

public function setModulopertencente($modulopertencente){
 $this->modulopertencente=$modulopertencente;
}

public function getDisciplrequisito(){
return $this->disciplrequisito;
}

public function setDisciplrequisito($disciplrequisito){
 $this->disciplrequisito=$disciplrequisito;
}

public function getId_plano_ensino(){
return $this->id_plano_ensino;
}

public function setId_plano_ensino($id_plano_ensino){
 $this->id_plano_ensino=$id_plano_ensino;
}

public function getId_professor(){
return $this->id_professor;
}

public function setId_professor($id_professor){
 $this->id_professor=$id_professor;
}
public function equals($object){
if($object instanceof Disciplina){

if($this->id_disciplina!=$object->id_disciplina){
return false;

}

if($this->nome!=$object->nome){
return false;

}

if($this->disciplsigla!=$object->disciplsigla){
return false;

}

if($this->creditos!=$object->creditos){
return false;

}

if($this->descricao!=$object->descricao){
return false;

}

if($this->cargahoraria!=$object->cargahoraria){
return false;

}

if($this->modulopertencente!=$object->modulopertencente){
return false;

}

if($this->disciplrequisito!=$object->disciplrequisito){
return false;

}

if($this->id_plano_ensino!=$object->id_plano_ensino){
return false;

}

if($this->id_professor!=$object->id_professor){
return false;

}

return true;

}
else{
return false;
}

}
public function toString(){

 return "  [id_disciplina:" .$this->id_disciplina. "]  [nome:" .$this->nome. "]
           [disciplsigla:" .$this->disciplsigla. "]  [creditos:" .$this->creditos. "]
           [descricao:" .$this->descricao. "]  [cargahoraria:" .$this->cargahoraria. "]
           [modulopertencente:" .$this->modulopertencente. "]
           [disciplrequisito:" .$this->disciplrequisito. "]
           [id_plano_ensino:" .$this->id_plano_ensino. "]
           [id_professor:" .$this->id_professor. "]  " ;
}

}

 ?>

File: GestaoAcademicaWeb/Madodis/src/br/edu/facol/gestaoacademicaweb/pojo/Disciplina.java
package br.edu.facol.gestaoacademicaweb.pojo;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.validation.Valid;
import javax.validation.constraints.Max;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@Entity
@Table(name="TB_DISCIPLINA")
public class Disciplina extends BaseObject {

	@NotNull(message="Nome não pode ser nulo ou vazio.")
	@Size(min=1, max=100, message="Nome deve ter entre 1 e 100 caracteres.")
	@Column(name="NOME")
	private String nome;
	
	@NotNull(message="Sigla não pode ser nulo ou vazio.")
	@Size(min=1, max=100, message="Sigla deve ter entre 1 e 100 caracteres.")
	@Column(name="SIGLA")
	private String sigla;
	
	@Max(value=40, message="Os creditos máximos são 40 horas")
	@Min(value=20, message="Os creditos minimos são 20 horas")
	@Column(name="CREDITOS")
	private int creditos;
	
	@Max(value=120, message="A carga horaria máxima são 120 horas")
	@Min(value=35, message="A carga horaria minima são 35 horas")
	@Column(name="CARGA_HORARIA")
	private int cargaHoraria;
	
	@Max(value=10, message="O modulo pertencente máximos é 10")
	@Min(value=1, message="O modulo pertencente minimo é 1")
	@Column(name="MODULO_PERTENCENTE")
	private int moduloPertencente;
	
	@NotNull(message="Descrição nãoo pode ser nulo ou vazio.")
	@Size(min=15, max=250, message="Descriçãoo deve ter entre 15 e 250 caracteres.")
	@Column(name="DESCRICAO")
	private String descricao;
	
	@NotNull(message="Requisitos não pode ser nulo ou vazio.")
	@Size(min=10, max=250, message="Requisitos deve ter entre 10 e 250 caracteres.")
	@Column(name="REQUISITOS")
	private String requisitos;
	
	@OneToOne(fetch=FetchType.EAGER)
	private Professor professor;
	
	@Valid
	@OneToOne(cascade=CascadeType.ALL, fetch=FetchType.EAGER)
	private PlanoDeEnsino planoDeEnsino;
	
	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome;
	}
	
	public String getDescricao() {
		return descricao;
	}

	public void setDescricao(String descricao) {
		this.descricao = descricao;
	}
	
	public PlanoDeEnsino getPlanoDeEnsino() {
		return planoDeEnsino;
	}

	public void setPlanoDeEnsino(PlanoDeEnsino planoDeEnsino) {
		this.planoDeEnsino = planoDeEnsino;
	}

	public String getSigla() {
		return sigla;
	}

	public void setSigla(String sigla) {
		this.sigla = sigla;
	}

	public int getCreditos() {
		return creditos;
	}

	public void setCreditos(int creditos) {
		this.creditos = creditos;
	}

	public int getCargaHoraria() {
		return cargaHoraria;
	}

	public void setCargaHoraria(int cargaHoraria) {
		this.cargaHoraria = cargaHoraria;
	}

	public int getModuloPertencente() {
		return moduloPertencente;
	}

	public void setModuloPertencente(int moduloPertencente) {
		this.moduloPertencente = moduloPertencente;
	}

	public String getRequisitos() {
		return requisitos;
	}

	public void setRequisitos(String requisitos) {
		this.requisitos = requisitos;
	}

	public Professor getProfessor() {
		return professor;
	}

	public void setProfessor(Professor professor) {
		this.professor = professor;
	}

	@Override
	public String toString() {
		return " Disciplina [ id = " + getId() + ", nome = " + getNome() + "]";
	}

}
